rpc/client.go

```diff
@@ -41,11 +45,16 @@ type response[T serde.ResponsePayload] struct {
 
 // NewRpcClient creates a new RpcClient
 func NewRpcClient(rpcServerUrl string, logger zerolog.Logger, trans transport.Requester) (*RpcClient, error) {
-	c := &RpcClient{trans, logger, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}}
-	err := c.subscribeToNotifications()
+	ctx, cancel := context.WithCancel(context.Background())
+	c := &RpcClient{trans, logger, &safesync.Map[chan struct{}]{}, &safesync.Map[chan query.LedgerChannelInfo]{}, &safesync.Map[chan query.PaymentChannelInfo]{}, cancel, sync.WaitGroup{}}
+
+	notificationChan, err := c.transport.Subscribe()
```

Comment:
By constructing the `notificationChan` here, we can run the function in a go-routine `go c.subscribeToNotifications(ctx, notificationChan)`, instead of nesting it within `subscribeToNotifications`